### DISCUSSION TITLE
Configuration file not in effect

### DISCUSSION THREAD

**Discussion by viitoare** [apache/airflow#40743] [General]
### Apache Airflow version

Other Airflow 2 version (please specify below)

### If "Other Airflow 2 version" selected, which one?

2.9.1

### What happened?

I am using the CeleryExecutor. When I set parallelism=300 only in airflow.cfg and do not add AIRFLOW__CORE__PARALLELISM in docker-compose.yaml, the value of self.parallelism printed in base_executor.py is 32. When I add AIRFLOW__CORE__PARALLELISM: 300 in docker-compose.yaml, the value of self.parallelism printed in base_executor.py is the configured 300.

### What you think should happen instead?

It should only take effect when I configure parallelism in airflow.cfg, rather than setting AIRFLOW_ CORE.PLALLELISM in Docker Compose.yaml for it to take effect

### How to reproduce

Only configure parallelism=300 in airflows.cfg, do not set AIRFLOW__CORE__PARALLELISM in docker-compose. yaml, and then print the value of self. parallelism in base_deecutor. py. You can see that the printed value is 32
![微信截图_20240710192422](https://github.com/apache/airflow/assets/130271253/5b396816-cc55-452b-8753-35f2908f34b4)
![微信截图_20240710192444](https://github.com/apache/airflow/assets/130271253/a3bab3cd-a4ab-4671-b94c-bad6a2a6d173)

### Operating System

Kylin Linux Advanced Server v10

### Versions of Apache Airflow Providers

2.9.1

### Deployment

Docker-Compose

### Deployment details

Celery Version 5.4.0
Docker Compose version v2.5.0

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by shahar1** [comment]
Could you please edit the content of "How to reproduce" section and translate it into English?

**Comment by viitoare** [comment]
> Could you please edit the content of "How to reproduce" section and translate it into English?

Sorry, it was my mistake and it has been corrected

**Comment by potiuk** [comment]
I think you set it in a different/wrong file.  The docker-compose of ours is really starting point for people to get it and modify it on their own for example here is the comment explaining what to do. when you want to use your own config for experimentation. Especially it is important that you realise that the config file inside some of the containers run are not the same as config files on the host.

For example see that file:

```
    # The following line can be used to set a custom config file, stored in the local config folder
    # If you want to use it, outcomment it and replace airflow.cfg with the name of your config file
    # AIRFLOW_CONFIG: '/opt/airflow/config/airflow.cfg'
```

I believe this is some wrong assumption you made about which config file needs to be modified - but I am guessing. Converting it into discussion - you should take a look and confirm (explain) how, where you modified the config. How you made it actually being used in the containers and how you checked that actually the config file you modified is present and used inside the container that you get the logs from.

**Comment by viitoare** [comment]
[airflow.txt](https://github.com/user-attachments/files/16196194/airflow.txt)
[docker-compose.txt](https://github.com/user-attachments/files/16196195/docker-compose.txt)
I don't think there's a problem with my configuration, it works fine, it's just that parallelism is not the value I set

**Comment by potiuk** [comment]
CAn you explain how you set it in the config and how your config finds the way into the container?

**Reply by potiuk** [reply]
Also - maybe you ALSO have the env variable set regardless ? setting env variable or secret manager overrides what is set in the config

**Reply by viitoare** [reply]
> CAn you explain how you set it in the config and how your config finds the way into the container?

After downloading the airflow image, create an airflow directory, then create dags, plugins and other directories in the airflow directory, put the modified airflow.cfg file and docker-compose.yaml file in the airflow directory, only set parallelism in airflow.cfg, delete AIRFLOW__CORE__PARALLELISM in docker-compose.yaml, and there is no other setting. The value of parallelism printed here is 32, but when I set AIRFLOW__CORE__PARALLELISM in docker-compose.yaml, the value of parallelism printed is the value I set

**Reply by potiuk** [reply]
> After downloading the airflow image, create an airflow directory, then create dags, plugins and other directories in the airflow directory,

So you did not folllow this chapter https://airflow.apache.org/docs/apache-airflow/stable/howto/docker-compose/index.html#special-case-adding-a-custom-config-file  did you?

**Reply by potiuk** [reply]
The reason why I am asking the question is because I cannot reproduce it:

```
oot@1dfbe165f06e:/opt/airflow# cat /root/airflow/airflow.cfg  | grep parallelism
parallelism = 300
# This should not be greater than ``[core] parallelism``.
# Set this to 0 to use the value of ``[core] parallelism``
sync_parallelism = 0
root@1dfbe165f06e:/opt/airflow# set| grep PARAL
root@1dfbe165f06e:/opt/airflow# airflow scheduler
  ____________       _____________
 ____    |__( )_________  __/__  /________      __
____  /| |_  /__  ___/_  /_ __  /_  __ \_ | /| / /
___  ___ |  / _  /   _  __/ _  / / /_/ /_ |/ |/ /
 _/_/  |_/_/  /_/    /_/    /_/  \____/____/|__/
[2024-07-12T15:59:19.632+0000] {task_context_logger.py:66} INFO - Task context logging is enabled
[2024-07-12T15:59:19.633+0000] {base_executor.py:131} INFO - Paralelism: 300
[2024-07-12T15:59:19.633+0000] {executor_loader.py:272} INFO - Loaded executor: LocalExecutor
```

And I think your problem is that you have not make the config available to inside the container (and I already explained it before)

**Reply by viitoare** [reply]
> > After downloading the airflow image, create an airflow directory, then create dags, plugins and other directories in the airflow directory,
> 
> So you did not folllow this chapter https://airflow.apache.org/docs/apache-airflow/stable/howto/docker-compose/index.html#special-case-adding-a-custom-config-file did you?

I just added the file mapping - ./airflow.cfg:/opt/airflow/airflow.cfg in docker-compose.yaml, and did not set AIRFLOW_CONFIG: '/opt/airflow/config/airflow.cfg'

**Reply by viitoare** [reply]
> The reason why I am asking the question is because I cannot reproduce it:
> 
> ```
> oot@1dfbe165f06e:/opt/airflow# cat /root/airflow/airflow.cfg  | grep parallelism
> parallelism = 300
> # This should not be greater than ``[core] parallelism``.
> # Set this to 0 to use the value of ``[core] parallelism``
> sync_parallelism = 0
> root@1dfbe165f06e:/opt/airflow# set| grep PARAL
> root@1dfbe165f06e:/opt/airflow# airflow scheduler
>   ____________       _____________
>  ____    |__( )_________  __/__  /________      __
> ____  /| |_  /__  ___/_  /_ __  /_  __ \_ | /| / /
> ___  ___ |  / _  /   _  __/ _  / / /_/ /_ |/ |/ /
>  _/_/  |_/_/  /_/    /_/    /_/  \____/____/|__/
> [2024-07-12T15:59:19.632+0000] {task_context_logger.py:66} INFO - Task context logging is enabled
> [2024-07-12T15:59:19.633+0000] {base_executor.py:131} INFO - Paralelism: 300
> [2024-07-12T15:59:19.633+0000] {executor_loader.py:272} INFO - Loaded executor: LocalExecutor
> ```
> 
> And I think your problem is that you have not make the config available to inside the container (and I already explained it before)

I will try AIRFLOW_CONFIG: '/opt/airflow/config/airflow.cfg' next week. Thanks.

**Reply by potiuk** [reply]
OK. So now we are onto something. You also added mapping then (this is what I was asking for). default place where airflow looks for config if AIRFLOW_CONFIG is not set, is AIRFLOW_HOME/airflow.cfg which in your case is lkely /root/airflow/airflow.cfg.

**Reply by potiuk** [reply]
Following the docs is good idea generally in this case.

**Comment by viitoare** [comment]
It seems that the permissions on my airflow.cfg file are the issue. I don't know why my file permissions are set to 600, but changing them to 644 made the configuration work,Thanks!
